src/ray/raylet/node_manager.cc

```diff
@@ -2916,6 +2916,13 @@ MemoryUsageRefreshCallback NodeManager::CreateMemoryUsageRefreshCallback() {
               rpc::WorkerExitType::USER_ERROR,
               absl::StrCat("Ray OOM killer terminating worker to prevent system OOM"),
               true /* force */);
+          if (latest_worker->GetActorId().IsNil()) {
+            ray::stats::STATS_memory_manager_worker_eviction_total.Record(
+                1, "MemoryManager.TaskEviction.Total");
+          } else {
+            ray::stats::STATS_memory_manager_worker_eviction_total.Record(
```

Comment:
Actually should we avoid killing actor if possible? Actors are not restartable by default, which means killing it will likely fail the workload. We should probably only kill it when max_restart is not 0.

Reply:
We must always kill actors. The primary goal here is to protect the system processes from user code. User code that is OOMing will fail no matter what.

Reply:
Agree that the user code that's OOMing will fail no matter what. The issue now is that we are not killing the worker that's causing OOM but the latest submitted task.

Reply:
Also there are tasks that just need a bit more memory to finish successfully. Like they will succeed if we don't run too many tasks on the same node in the meantime.

Reply:
Let's move the policy discussion to release-test where we will look at the effectiveness of the oom killer against our current benchmarks.

Please evaluate this PR based on observability / logging. From my perspective i think it is valid to OOM kill an actor and so we should add counter for it. The policy for when / if to kill an actor is something we will need to resolve as part of the policy discussion